dashboard/src/components/Filter/CodeBlock.tsx

```diff
@@ -74,16 +132,23 @@ const CodeBlock = ({
 
   return (
     <div>
-      <pre
-        className={cn(
-          'max-h-[275px] w-full max-w-[calc(100vw_-_398px)] overflow-x-auto bg-[#DDDDDD] p-4 font-mono text-[#767676]',
-          className,
-        )}
-      >
-        <div
-          dangerouslySetInnerHTML={{ __html: highlightedCode.highlightedCode }}
-        ></div>
-      </pre>
+      <div className="py-4 pl-3">
+        <span className="font-bold">
+          <FormattedMessage
+            id="buildAccordion.logExcerpt"
+            defaultMessage="Log Excerpt"
+          />
+        </span>
+      </div>
+
+      <CodeBlockDialog
+        code={
+          <MemoizedCode
+            className={className}
+            highlightedCode={highlightedCode.highlightedCode}
+          />
+        }
+      />
       {highlightedCode.highlightCount > 0 && (
```

Comment:
Would it be interesting to extract the highlight counter to a separate component and show it when the log is in fullscreen as well?